### ISSUE TITLE
Documentation improvement: full-width responsive, with gutter and without side-margins.

### ISSUE THREAD

**Issue by RichardNeill** [twbs/bootstrap#33219]
May I suggest another example, that would be helpful here: https://getbootstrap.com/docs/4.1/layout/grid/

Namely, what is the right way to get a full-width responsive 2+-column layout, with gutter **between** the columns, but not **around** them. This is, I think, a fairly common use-case requirement, and it's not obvious (to me anyway) from the docs as to how to achieve it.

Representing the edge of the screen by "[","]",  content as "aaa", and whitespace as "."

On Mobile, we want the cols stacked, :

[aaaaaaaaa]
[bbbbbbbbb]

Whereas on Desktop:

[aaaaaaaaa...bbbbbbbbb]

i.e. there is a gutter between aaa, bbb  which automatically appears when the columns reflow, but the left-col remains hard-to-the-left, and the right-col remains hard-to-the-right. Similarly we could have:
 [aaaaaaaaa...bbbbbbbbb...ccccccccc...ddddddddd]

The docs don't show how to achieve this. They do explain 3 other options though:

* wrapping it in container-fluid results in margins at the side (which I am asking how to eliminate):

[...aaaaaaaaa...bbbbbbbbb...]

* using .no-gutter results in no internal spaces:

[aaaaaaaaabbbbbbbbb]

* omitting the container results in an unwanted horizontal scrollbar:

[aaaaaaaaa...bbbbbbbbb...]

**Comment by aribudin** [comment]
Hello
I think this possible with d-flex

```
<div class="d-flex flex-column flex-lg-row justify-content-center justify-content-lg-between bg-primary mb-3">
    <div class="custom-column p-2 flex-fill bg-warning">Item here</div>
    <div class="custom-column p-2 flex-fill bg-warning">Item here</div>
    <div class="custom-column p-2 flex-fill bg-warning">Item here</div>
</div>
```
Dont forget to add space in lg screen up with this scss

```
@include media-breakpoint-up(lg) { 
 .custom-column {
   max-width: 32%;
}
}
```

**Comment by ffoodd** [comment]
@RichardNeill Would you mind providing either a visual example or a sample test case that does what you'd want, please?
Not really easy to understand, for now.

**Comment by RichardNeill** [comment]
Hello. I've found a way to do this:

&lt;div.container-fluid px-0&gt;
    &lt;div class='row no-gutters'&gt;
      &lt;div class='col-xl-6 pr-xl-4'&gt;Section A&lt;/div&gt;
     &lt;div class='col-xl-6 pl-xl-4'&gt;Section B&lt;/div&gt;
   &lt;div&gt;
&lt;div&gt;

This does what I want, i.e.

* For smaller screens, stack section A and section B,  each being full width.
* For large screens, 2 column layout, with a gap between A and B, but  A has no left-margin, B has no right-margin, and there is no unwanted horiz scrollbar.

So...
1. I think this approach might be worth documenting.
2. It is rather ugly, because of the need to explicitly add the pr and pl as the columns re-wrap. It would get progressively nastier if we had more than 2 cols, and more than 2 types of screen to support.

What I think would be better would be to have a "row.only-inner-gutters" class that would keep the gutters between the columns, but not around them.  The effect of .only-inner-gutters would be:

* add padding on L and R
* but not to the first-child or last-child.
* and not to the children that get a soft-wrap.

Thanks!

**Comment by donquixote** [comment]
Here is a solution idea how the horizontal scrollbar can be avoided: #33911.

**Comment by alpadev** [comment]
@ffoodd does this explain it?
![grid-example](https://user-images.githubusercontent.com/2838324/117646201-c6dee900-b18b-11eb-8179-0f9cc91abf28.png)

**Comment by alpadev** [comment]
He had a problem with understanding the description and asked for an visual example. That's what I did.. I have no doubts he's capable of finding a solution (unlike others here maybe..)

**Comment by mdo** [comment]
This will be solved by our CSS grid implementation: see #31813. Using utilities you can get something close to this already though: https://codepen.io/emdeoh/pen/OJpMaxr.

**Comment by mdo** [comment]
If your intended browser support works for it, you can also use the `gap` utilities and properties on flex containers like `.row`.

```html
<div class="row g-0 gap-3">
  <div class="col">Left</div>
  <div class="col">Right</div>
</div>
```

Closing out with this comment and the above.
